### PR TITLE
Add configuration section

### PR DESCRIPTION
**Pull request by soywiz**

### CODE REVIEW

File: pages/servers/configuration.md [hunk 1]
```diff
@@ -3,7 +3,198 @@ title: Configuration
...
+    var configureBootstrap: ServerBootstrap.() -> Unit = {} // User-provided function to configure Netty's [ServerBootstrap]
+    var responseWriteTimeoutSeconds: Int = 10 // Timeout in seconds for sending responses to client
+}
+```
+
+### Jetty
+
+```kotlin
+class Configuration : BaseApplicationEngine.Configuration() {
+    var configureServer: Server.() -> Unit = {}
```
Comment (cy6erGn0m) [inline pages/servers/configuration.md]: Property to provide lambda that will be called during Jetty server initialization with server instance as argument

File: pages/servers/configuration.md [hunk 2]
```diff
@@ -3,7 +3,198 @@ title: Configuration
...
+}
+```
+
+### CIO
+
+```kotlin
+class Configuration : BaseApplicationEngine.Configuration() {
+    var connectionIdleTimeoutSeconds: Int = 45
```
Comment (cy6erGn0m) [inline pages/servers/configuration.md]: `connectionIdleTimeoutSeconds` number of seconds while server will keep HTTP IDLE connections open. A connection is IDLE if there are no active requests running

File: pages/servers/configuration.md [hunk 3]
```diff
@@ -3,7 +3,198 @@ title: Configuration
...
+}
+    
+```
+
+### Tomcat
+
+```kotlin
+class Configuration : BaseApplicationEngine.Configuration() {
+    var configureTomcat: Tomcat.() -> Unit = {}
```
Comment (cy6erGn0m) [inline pages/servers/configuration.md]: Similar to Jetty but for Tomcat

File: pages/servers/configuration.md [hunk 4]
```diff
@@ -3,7 +3,198 @@ title: Configuration
...
+```kotlin
+val moduleName = Application::module.javaMethod!!.let { "${it.declaringClass.name}.${it.name}" }
+io.ktor.server.netty.main(args + arrayOf("-port=8080", "-PL:ktor.application.modules=$moduleName"))
+```
+
+Or with a custom `applicationEngineEnvironment`:
+
+```kotlin
+embeddedServer(Netty, applicationEngineEnvironment {
+    this.log = LoggerFactory.getLogger("ktor.application")
```
Comment (cy6erGn0m) [inline pages/servers/configuration.md]: Not sure about `this.` explicitly specified. Wouldn't it look better without it? especially DSL for `module {}` and `this.port = 8080`

**Comment by soywiz** [comment]
Addressed all the issues.

This still requires @B1ggDave proofreading.

File: pages/servers/configuration.md [hunk 5]
```diff
@@ -20,8 +20,8 @@ This format is similar to JSON, but optimized for being read and written by huma
 <a id="embedded-server"></a>
 ## Configuring embeddedServer
 
-`embeddedServer` includes an optional parameter `configure` specific per factory where you can set generic
-and specific configuration. Any configuration will have these common available fields to set:
+`embeddedServer` includes an optional parameter `configure` specific per factory where you can set a generic
```
Comment (soywiz) [inline pages/servers/configuration.md]: Indeed, *specific per factory* maybe is rather confusing to the user too.

It happens that the first parameter `embeddedServer` is called `factory` and it is an `ApplicationEngineFactory`. It allows to construct an `ApplicationEngine` and has associated a specific `ApplicationEngine.Configuration`.

What about:
> `embeddedServer` includes an optional parameter `configure` that allows you to set the associated configuration to the `ApplicationEngineFactory` specified in the first parameter. All the configurations will have some common properties that you can set:

Comment (B1ggDave) [reply pages/servers/configuration.md]: Nice yes, your suggestion works well!

**Pull request closed without merging**